client/pages/tables.jsx

```diff
@@ -0,0 +1,131 @@
+// import Redirect from '../components/redirect';
+import Navbar from '../components/navbar';
+import ListTables from '../components/listTables';
+
+export default class Tables extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      tablesFetch: [],
+      isLoading: true,
+      isFreeModalSelected: false,
+      isAssignModalSelected: false,
+      tableSelected: null
+    };
+    this.freeTable = this.freeTable.bind(this);
+    this.assignTable = this.assignTable.bind(this);
+    this.noBtnClick = this.noBtnClick.bind(this);
+    this.yesBtnClick = this.yesBtnClick.bind(this);
+
+  }
+
+  componentDidMount() {
+    fetch('/api/get/tables')
+      .then(res => res.json())
+      .then(result => {
+        this.setState({ tablesFetch: result, isLoading: false });
+      });
+  }
+
+  freeTable(event) {
+    this.setState({ isFreeModalSelected: true, tableSelected: event.target.id });
+  }
+
+  assignTable(event) {
+    this.setState({ isAssignModalSelected: true, tableSelected: event.target.id });
+  }
+
+  noBtnClick(event) {
+    this.setState({ [event.target.name]: false, tableSelected: null });
+  }
+
+  yesBtnClick(event) {
+    if (event.target.name === 'isFreeModalSelected') {
+      const req = {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: null
+      };
+      fetch(`/api/put/table/${this.state.tableSelected}`, req)
+        .then(res => res.json())
+        .then(result => {
+        });
+      for (let i = 0; i < this.state.tablesFetch.length; i++) {
+        if (this.state.tablesFetch[i].tableNumber === parseInt(this.state.tableSelected)) {
+          const newdata = this.state.tablesFetch.slice(i, i + 1);
+          newdata[0].userId = null;
+          const newstate = this.state.tablesFetch.slice(0, i).concat(newdata[0]).concat(this.state.tablesFetch.slice(i + 1));
+          this.setState({ tablesFetch: newstate, isFreeModalSelected: false, tableSelected: null });
+        }
+      }
+    } if (event.target.name === 'isAssignModalSelected') {
+      const req = {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({ custId: this.props.userId })
+      };
+      fetch(`/api/put/table-assign/${this.state.tableSelected}`, req)
+        .then(res => res.json())
+        .then(result => {
+        });
+      for (let i = 0; i < this.state.tablesFetch.length; i++) {
+        if (this.state.tablesFetch[i].tableNumber === parseInt(this.state.tableSelected)) {
+          const newdata = this.state.tablesFetch.slice(i, i + 1);
+          newdata[0].userId = this.props.userId;
+          const newstate = this.state.tablesFetch.slice(0, i).concat(newdata[0]).concat(this.state.tablesFetch.slice(i + 1));
+          this.setState({ tablesFetch: newstate, isAssignModalSelected: false, tableSelected: null });
+        }
```

Comment:
this code is mostly duplicated. The difference that I can find is that the url path is different and the body is different. Here are some things you can do to cut that down a little bit.

- use a condition to choose the correct url path
- use a condition to choose the correct body
- send the request
  - then, update the state (inside the fetch promise callback)